internal/cloud/keyvault.go

```diff
@@ -55,134 +53,103 @@ func getAzureClientAuthMethod() clientAuthMethod {
 	return mode
 }
 
-// NewAzureClient returns a new Azure Key Vault client
-// By default, the authorizer will be from Managed Identity
-// If user sets AKV_AUTH_METHOD to AKV_AUTH_FROM_CLI, it will use Azure CLI instead
-func NewAzureClient() (*keyvault.BaseClient, error) {
-	var (
-		authorizer autorest.Authorizer
-		err        error
-	)
+// AzureCredential returns an Azure TokenCredential
+func AzureCredential() (credential azcore.TokenCredential, err error) {
```

Comment:
This function is not called outside of this package. We can make it private.

Reply:
Updated.